File: java/src/main/java/deltix/luminary/implementation/ItemNode.java
package deltix.luminary.implementation;

import java.util.ArrayList;
import java.util.List;

abstract class ItemNode {
    private final String name;
    private final List<DecoratorValueNode> decorators = new ArrayList<>();
    private final List<String> comments;

    protected ItemNode(String name, List<String> comments) {
        this.name = name;
        this.comments = comments != null ? comments : new ArrayList<String>();
    }

    public String getName() {
        return name;
    }

    public List<DecoratorValueNode> getDecorators() {
        return decorators;
    }

    public List<String> getComments() {
        return comments;
    }
}

File: csharp/main/Implementation/ItemNode.cs
using System;
using System.Collections.Generic;
using JetBrains.Annotations;

namespace Deltix.Luminary.Implementation
{
	internal abstract class ItemNode
	{
		protected ItemNode([NotNull] String name, [CanBeNull] List<String> comments)
		{
			Name = name;
			Comments = comments ?? new List<String>();
		}

		[NotNull]
		public String Name { get; }

		[NotNull]
		public List<DecoratorValueNode> Decorators { get; } = new List<DecoratorValueNode>();

		[NotNull]
		public List<String> Comments { get; }
	}
}
